[power_management] docs for new component that enables power management for esp_idf

rwrozelle opened the pull request:
## Description:

This is a new component power_management that enables esp-idf power management.
See https://docs.espressif.com/projects/esp-idf/en/stable/esp32/api-reference/system/power_management.html

**Related issue (if applicable):** fixes <link to issue>

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** 

- esphome/esphome#12325

## Checklist:

  - [x] I am merging into `next` because this is new documentation that has a matching pull-request in [esphome](https://github.com/esphome/esphome) as linked above.  
    or
  - [ ] I am merging into `current` because this is a fix, change and/or adjustment in the current documentation and is not for a new component or feature.

  - [ ] Link added in `/components/index.rst` when creating new documents for new components or cookbook.

<details>
<summary><strong>New Component Images</strong></summary>

If you are adding a new component to ESPHome, you can automatically generate a standardized black and white component name image for the documentation.

**To generate a component image:**

1. Comment on this pull request with the following command, replacing `COMPONENT_NAME` with your component name in **UPPER_CASE** format with **underscores** (e.g., `BME280`, `SHT3X`, `DALLAS_TEMP`):

   ```
   @esphomebot generate image COMPONENT_NAME
   ```

2. The ESPHome bot will respond with a downloadable ZIP file containing the SVG image.

3. Extract the SVG file and place it in the `images/` folder of this repository.

4. Use the image in your component's index table entry in `/components/index.rst`.

**Example:** For a component called "DHT22 Temperature Sensor", use:
```
@esphomebot generate image DHT22
```

</details>

rwrozelle committed:
Create power_management.md

rwrozelle committed:
Create power_management.md

rwrozelle committed:
Merge branch 'power_management1' of https://github.com/rwrozelle/esphome-docs into power_management1

rwrozelle committed:
Lint Fixes

rwrozelle committed:
fix yaml

rwrozelle committed:
fix config-time link

rwrozelle committed:
add image

rwrozelle committed:
typo

rwrozelle committed:
updated

rwrozelle committed:
add switch

rwrozelle committed:
typo

rwrozelle committed:
Rename initial_lock_duration to timer_lock_durationfi

rwrozelle committed:
complex lock actions

rwrozelle committed:
lint fixes

rwrozelle committed:
Merge remote-tracking branch 'upstream/next' into power_management1

rwrozelle committed:
Simplified

rwrozelle committed:
Fix typos

rwrozelle committed:
document OTA

rwrozelle committed:
remove ota recommendations

rwrozelle committed:
Merge remote-tracking branch 'upstream/next' into power_management1

rwrozelle committed:
Merge remote-tracking branch 'upstream/next' into power_management1

rwrozelle committed:
Add esphome_locks

rwrozelle committed:
fix typo

rwrozelle committed:
add better documentation

rwrozelle committed:
Merge remote-tracking branch 'upstream/next' into power_management1

rwrozelle committed:
typo

rwrozelle committed:
expanded discussion

rwrozelle committed:
Merge remote-tracking branch 'upstream/next' into power_management1

rwrozelle committed:
Merge remote-tracking branch 'upstream/next' into power_management1

rwrozelle committed:
default power_down_peripherals to True

rwrozelle committed:
fix power_down_peripherals to be variant based

rwrozelle committed:
fix typo

rwrozelle committed:
Merge remote-tracking branch 'upstream/next' into power_management1

rwrozelle committed:
add explanation of task notification

rwrozelle committed:
Merge remote-tracking branch 'upstream/next' into power_management1

rwrozelle committed:
Update light sleep explanation in power_management.md

Clarified conditions for light sleep in power management.

rwrozelle committed:
Merge remote-tracking branch 'upstream/next' into power_management1

Unfocused commented on src/content/docs/components/power_management.md:
```diff
@@ -0,0 +1,152 @@
+
+Using power_management component comes at the cost of increased interrupt latency and can be upwards of 40 us.
+
+When power_down_flash: True, device will try to power down flash when entering Light Sleep, which costs more time when device wakes up. Can only be enabled if there is no SPIRAM configured.
```
The documentation consistently refers to this as PSRAM, never SPIRAM.

rwrozelle replied:
done

rwrozelle committed:
Fix typo in power management documentation

Corrected 'SPIRAM' to 'PSRAM' for accuracy in documentation regarding power management features.

rwrozelle committed:
Merge remote-tracking branch 'upstream/next' into power_management1
